src/engine/blocks.js

```diff
@@ -379,6 +379,9 @@ class Blocks {
                 if (!editingTarget.lookupVariableById(e.varId)) {
                     editingTarget.createVariable(e.varId, e.varName, e.varType);
                     this.emitProjectChanged();
+                    // TODO this should probably be batched
+                    // (esp. if we receive multiple new var_creates in a row).
+                    this.runtime.requestToolboxExtensionsUpdate();
```

Comment:
If we end up changing the gui handling of the variable create button so that it no longer goes through scratch-blocks to display the variable creation modal (and triggering the actual variable creation) this code should be removed from here and moved into the new extension function that will be responsible for triggering variable creation.

Reply:
Does this cause a significant performance problem when loading a project which uses a lot of variables?

Reply:
We talked about this one offline. I think this code path should not be called during project load, but I will confirm. I'll also look at flame charts with @picklesrus.

Reply:
I locally removed the old variables category (so that only the extension was loaded, and I loaded a new project containing variables, and confirmed that this code path is not called during project load, and the variable blocks still show up in the toolbox correctly).